src/Features/Core/Portable/Completion/CompletionService_GetCompletions.cs

```diff
@@ -88,6 +88,7 @@ public abstract partial class CompletionService
             // PERF: Many CompletionProviders compute identical contexts. This actually shows up on the 2-core typing test.
             // so we try to share a single SyntaxContext based on document/caretPosition among all providers to reduce repeat computation.
             var sharedContext = new SharedSyntaxContextsWithSpeculativeModel(document, caretPosition);
+            _ = sharedContext.GetSyntaxContextAsync(document, cancellationToken);
```

Comment:
what does thsi do?  doc why you're doing this here.

Reply:
This starts the calculation of SyntaxContext in the background so it will be available when providers access it. Will add a comment

Reply:
skeptical this is necessary.  aren't we going to call into a provider virtually immediately?  so the first provider that calls will kick this off, and hte rest of hte providers will get that same asynclazy computation.